backend/vault/vault.go

```diff
@@ -0,0 +1,44 @@
+package vault
+
+import (
+	"context"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/heetch/confita/backend"
+)
+
+type vaultAPI interface {
+	Read(p string) (*api.Secret, error)
+}
+
+// Backend loads keys from Vault
+type Backend struct {
+	client vaultAPI
+	path   string
+}
+
+// NewBackend creates a configuration loader that loads from Vault
+func NewBackend(c *api.Logical, p string) *Backend {
+	return &Backend{
+		client: c,
+		path:   p,
+	}
+}
+
+// Get loads the given key from Vault
+func (b *Backend) Get(ctx context.Context, key string) ([]byte, error) {
+	secret, err := b.client.Read(b.path)
+	if err != nil {
+		return nil, err
+	}
+
+	if secret == nil {
+		return nil, backend.ErrNotFound
```

Comment:
I don't think we should return `backend.ErrNotFound` when the secret key is nil since the error is not caused by the given `key` param but by the `path` field. We should return a custom error instead IMO, something like `fmt.Errorf("unable to get secret from path \"%s\"", b.path)`

Reply:
Yeah I thought like you until I saw that the vault cli client prints "value not found" when the secret is wrong. 
But in our case, I think we should do like you are saying but for the error content I suggest the following:
`fmt.Errorf("secret not found at the following path: %s", b.path)`.
What do you think ?